### ISSUE TITLE
DefaultTypeResolverBuilder#_subtypeValidator causes NPE with JsonTypeInfo.Id.CLASS or MINIMAL_CLASS

### ISSUE THREAD

**Issue by egetman** [FasterXML/jackson-databind#2472]
Hi guys!
I faced with an issue with last update (2.10.0.pr3):
There was added a DefaultTypeResolverBuilder constructor with PolymorphicTypeValidator, and the @ Deprecated one passes the null reference by default. 
```java
    @Deprecated // since 2.10
    public DefaultTypeResolverBuilder(DefaultTyping t) {
        this(t, null);
    }
```
So, in the cases of deserialization with `JsonTypeInfo.Id.CLASS` there will be NPE. For example the last redisson `JsonJacksonCodec` make the following configuration:
```java
protected void initTypeInclusion(ObjectMapper mapObjectMapper) {
        TypeResolverBuilder<?> mapTyper = new DefaultTypeResolverBuilder(DefaultTyping.NON_FINAL) {
            public boolean useForType(JavaType t) {
                switch (_appliesFor) {
                case NON_CONCRETE_AND_ARRAYS:
                    while (t.isArrayType()) {
                        t = t.getContentType();
                    }
                    // fall through
                case OBJECT_AND_NON_CONCRETE:
                    return (t.getRawClass() == Object.class) || !t.isConcrete();
                case NON_FINAL:
                    while (t.isArrayType()) {
                        t = t.getContentType();
                    }
                    // to fix problem with wrong long to int conversion
                    if (t.getRawClass() == Long.class) {
                        return true;
                    }
                    if (t.getRawClass() == XMLGregorianCalendar.class) {
                        return false;
                    }
                    return !t.isFinal(); // includes Object.class
                default:
                    // case JAVA_LANG_OBJECT:
                    return t.getRawClass() == Object.class;
                }
            }
        };
        mapTyper.init(JsonTypeInfo.Id.CLASS, null);
        mapTyper.inclusion(JsonTypeInfo.As.PROPERTY);
        mapObjectMapper.setDefaultTyping(mapTyper);
	rest of config...
```
NPE will be thrown in `StdTypeResolverBuilder#verifyBaseTypeValidity`, cause the `ptv` will be null.
So my question: is it possible (for backward compatibility) to change constructor in `DefaultTypeResolverBuilder` from
```java
    @Deprecated // since 2.10
    public DefaultTypeResolverBuilder(DefaultTyping t) {
        this(t, null);
    }
```
to
```java
    @Deprecated // since 2.10
    public DefaultTypeResolverBuilder(DefaultTyping t) {
        this(t, LaissezFaireSubTypeValidator.instance);
    }
```
?

**Comment by cowtowncoder** [comment]
Thank you for reporting this! I'll have a look later today; sounds like a legit problem and something we want to fix before 2.10.0 official.

**Comment by cowtowncoder** [comment]
This was a good catch: I added a test, defaults to avoid the issue, so should be good for official 2.10.0.
